### DISCUSSION TITLE
`cargo test` does not run `test_apply` ?

### DISCUSSION THREAD

**Discussion by udsamani** [dathere/qsv#455] [Q&A] [answered]
I have observed a weird thing and just wanted to make sure if anyone else is facing this too ?

When I just run `cargo test`, `test_apply` suite does not run at all ? Can someone try and let me know if it is the same for them ?

As soon as i remove `#[cfg(feature = "apply")]` , I am able to see `cargo test` running all tests. Shouldn't we always default running all tests with `cargo test` ?

**Comment by jqnatividad** [comment, marked as answer]
Hi @udsamani ,
To run all tests, be sure to set these features:

```
cargo test --features=apply,fetch,foreach,generate,lua,python,full

#or shorthand, use all_full which is equivalent to enabling all the features for the full binary variant
cargo test --features all_full
```

To run the `test_apply` suite specifically, at the very least, the `apply` feature needs to be enabled:

`cargo test test_apply --features apply`

And yes, all tests are indeed run in CI:

https://github.com/jqnatividad/qsv/blob/3b5ec48c9107acbfb88d66b60627cf64c19e0f1d/.github/workflows/rust.yml#L37-L38

This is all because we have four different qsv binary variants - qsv, qsvlite, qsvdp and qsvnp - and we have  different feature flags tp build each variant - `full`, `qsvlite`, `datapusher_plus` and `nopython_full`:

https://github.com/jqnatividad/qsv/blob/3b5ec48c9107acbfb88d66b60627cf64c19e0f1d/Cargo.toml#L19-L41

With `qsvnp` (np for "no python") being really `qsv` binary variant built with the `full` feature flag, but with the `python` feature disabled.

https://github.com/jqnatividad/qsv/blob/3b5ec48c9107acbfb88d66b60627cf64c19e0f1d/Cargo.toml#L176-L177

**Reply by udsamani** [reply]
Thank you for the detailed reply.
